docs/2023/README.md

```diff
@@ -0,0 +1,536 @@
+
+The structure of a `DeviceInviteAcceptance` (written to the invitee's chain) is:
+
+- The `ActionHash` of the KSR.
+- The `ActionHash` of the `DeviceInvite`.
+
+#### Device Invite API
+
+**Create**: The validation that happens when you create a new `DeviceInvite`:
+
+- A `DeviceInvite` must deserialize cleanly from the validating record.
+- The KSR must be fetched and deserialized into a `KeysetRoot`.
+- An invitee must have a different `AgentPubkey` than the invitor.
+- If the author of the invitation is the FDA in the invitation's KSR
+  - Do a hash-bounded query from the invite hash back to the KSR in the invitor's source chain.
+  - Check that that range contains no invite acceptances (have abandoned the Keyset they are inviting a new device into).
+- Else (author of invitation and FDA of KSR are not the same):
+  - Search from invite backwards (must_get_agent_activity of the invitor), find the first `DeviceInviteAcceptance` in their chain.
+  - The invite in that `DeviceInviteAcceptance` must fetch and deserialize to a `DeviceInvite`.
+  - That deserialized `DeviceInvite` must have the same KSR authority as the new `DeviceInvite` currently being validated.
+  - Also in that `DeviceInvite`, the invitee must be the author of the new `DeviceInvite`.
+
+We do not check whether the invitee exists on the DHT yet because they likely don't, that's why we're inviting them. If the `DeviceInviteAcceptance` is valid, and the `DeviceInvite` is valid, we trust that the parent's `DeviceInviteAcceptance` was properly validated, which ensures chain of authority to the KSR.
+
+**Read**: No direct read or lookup functions exposed in zome calls. The keyset tree structure is used internally for validation of key registration/revocation logic.
+
+**Update**: Not allowed.
+
+**Delete**: Not allowed.
+
+**Zome Calls**:
+
+- `invite_agent`
+  - Input is the `AgentPubKey` to invite.
+    - This agent does not exist on the DHT yet if they are planning to use the invite as their joining proof.
+  - Output is the exact `DeviceInviteAcceptance` the invitee must commit to their chain.
+  - Invites are always under the current keyset.
+
+#### Device Invite Acceptance API
+
+**Create**:
+
+- A `DeviceInviteAcceptance` must deserialize cleanly from the validating record
+- A `DeviceInvite` must be fetched and deserialize from the `invite` action hash on the `DeviceInviteAcceptance`
+- The author of the `DeviceInviteAcceptance` must be the referenced `AgentPubKey` on the `DeviceInvite`
+- The `KeysetRoot` must be the same on both the `DeviceInvite` and the `DeviceInviteAcceptance`
+
+**Read**: No exposed zome calls for read or lookup. For validation, the most recent `DeviceInviteAcceptance` is used to determine the current keyset.
+
+**Update**: Not allowed.
+
+**Delete**: Not allowed.
+
+##### Zome Calls
+
+- `accept_invite`
+  - input is a `DeviceInviteAcceptance`
+  - output is the `ActionHash` of the entry created
+  - creates the entry as-is from input
+
```

Comment:
Almost everything from here down was used in the rewrite.  The `README.md` covers the workflows at the highest level.  Then for more details about the architecture the reader can see the `INTEGRITY_MODEL.md` and `CONTRIBUTING.md`.  For usage and implementation details the reader can see the cargo documentation and examples in real javascript tests.